### PR TITLE
Fem: Create own base class for Python objects - fixes #14637

### LINKED ISSUE #14637: FEM: Some constraints are grayed out permanently in the tree view
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Problem description

Maybe I missed something in the recent smaller FEM changes but I've noticed that some constraints are always grayed out in the tree view (it can't be changed).

**0.22:**

**0.21.2:**

I could understand if those were all the constraints that don't have any symbols in the 3D view (it would still be confusing though as one can think the gray-out means they are inactive) but it's not the case (e.g. MaterialSolid is not grayed out).

@marioalexis84 Do you know anything about this?

### Full version info

```shell
OS: Windows 10 build 19045
Word size of FreeCAD: 64-bit
Version: 0.22.0dev.37709 (Git)
Build type: Release
Branch: main
Hash: dd816a635ccacc39f29b1761c6ccfed21a90b646
Python 3.11.9, Qt 5.15.13, Coin 4.0.2, Vtk 9.2.6, OCC 7.7.2
Locale: Polish/Poland (pl_PL)
```

### Subproject(s) affected?

FEM

### Anything else?

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by marioalexis84**
Create own base class for Python objects for future improvements. Also, this fixes #14637.

@FEA-eng

### CODE REVIEW

**Review by lyphrowny — COMMENTED** [review]
There is a "femviewprovider/view_base_femmaterial.py", but there is no "femobjects/base_femmaterial.py" - is it not needed?

I only reviewed the PR, didn't test it

File: src/Mod/Fem/femobjects/element_fluid1D.py [hunk 1]
```diff
@@ -63,14 +63,6 @@ class ElementFluid1D(base_fempythonobject.BaseFemPythonObject):
     def __init__(self, obj):
         super(ElementFluid1D, self).__init__(obj)
 
-        obj.addProperty(
-            "App::PropertyLinkSubList",
-            "References",
-            "FluidSection",
-            "List of fluid section shapes"
```
[the remark this answers is not in the document]
Comment (marioalexis84) [reply src/Mod/Fem/femobjects/element_fluid1D.py:70]: Yes, it is expected

File: src/Mod/Fem/femobjects/base_femmeshelement.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        prop.append(_PropHelper(
+            type  = "App::PropertyLinkSubList",
+            name  = "References",
+            group = "Mesh Element",
+            doc   = "List of reference shapes",
+            value = []
+            )
+        )
+
+        return prop
```
[the remark this answers is not in the document]
Comment (marioalexis84) [reply src/Mod/Fem/femobjects/base_femmeshelement.py:60]: Yes, it is for future improvement

File: src/Mod/Fem/femobjects/base_femmeshelement.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+## @package base_femmeshelement
+#  \ingroup FEM
+#  \brief base object for FEM Mesh Element Features
+
+
+from . import base_fempythonobject
+
+_PropHelper = base_fempythonobject._PropHelper
+
+class BaseFemMeshElement(base_fempythonobject.BaseFemPythonObject):
```
[the remark this answers is not in the document]
Comment (marioalexis84) [reply src/Mod/Fem/femobjects/base_femmeshelement.py:37]: We can change it after #14793 just so we don't rebase again.

**Comment by FEA-eng** [comment]
I was just going to test this 😅 Sorry for being too late to the show, I had no possibility to compile anything last week.

**Comment by chennes** [comment]
Well, you can still test it :) -- I only made sure it didn't break anything obvious (tests still ran, etc.), you are in a better position than me to do actual functional testing.

**Comment by FEA-eng** [comment]
Yeah, I compiled it anyway and I can confirm that it fixes the issue.

**Pull request merged**